### ISSUE TITLE
MineCraft 1.7.10 NoSuchMethod error

### ISSUE THREAD

**Issue by Sirse** [EssentialsX/Essentials#1498]
```
[11:50:46] [Server thread/ERROR] [Minecraft/]: Could not pass event PlayerInteractEvent to Essentials v2.0.1-b523
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:338) ~[JavaPluginLoader$1.class:1.7.10-1614.UNOFFICIAL]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[RegisteredListener.class:1.7.10-1614.UNOFFICIAL]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:507) [SimplePluginManager.class:1.7.10-1614.UNOFFICIAL]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:492) [SimplePluginManager.class:1.7.10-1614.UNOFFICIAL]
	at org.bukkit.craftbukkit.v1_7_R4.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:223) [CraftEventFactory.class:1.7.10-1614.UNOFFICIAL]
	at org.bukkit.craftbukkit.v1_7_R4.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:193) [CraftEventFactory.class:1.7.10-1614.UNOFFICIAL]
	at net.minecraft.network.NetHandlerPlayServer.func_147346_a(NetHandlerPlayServer.java:826) [nh.class:?]
	at net.minecraft.network.play.client.C08PacketPlayerBlockPlacement.func_148833_a(SourceFile:60) [jo.class:?]
	at net.minecraft.network.play.client.C08PacketPlayerBlockPlacement.func_148833_a(SourceFile:9) [jo.class:?]
	at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:245) [ej.class:?]
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:181) [nc.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:1023) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:432) [lt.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:841) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:693) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_131]
Caused by: java.lang.NoSuchMethodError: org.bukkit.entity.Player.getTargetBlock(Ljava/util/Set;I)Lorg/bukkit/block/Block;
	at com.earth2me.essentials.signs.SignPlayerListener.onSignPlayerInteract(SignPlayerListener.java:42) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor69.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:334) ~[JavaPluginLoader$1.class:1.7.10-1614.UNOFFICIAL]
	... 15 more
```

EssentialsX version: 201-b523

Server software: Thermos 1.7.10

**Comment by mdcfe** [comment]
Wasn't aware that 1.7.10 didn't have the new `Set<Material>` method yet. Might be a case for a provider, unless we're going to drop support for 1.7.10.

**Comment by mdcfe** [comment]
@SupaHam Thoughts on dropping support for 1.7.10?

@Sirse I don't think it's feasible for us to continue supporting 1.7.10 - Thermos hasn't been supported for over a year now, and 1.7.10 was released over three years ago. In that time, the Spigot-maintained Bukkit API has changed enough that 1.12+-friendly plugins will need to use awkward hacks to continue supporting ancient versions.

**Comment by SupaHam** [comment]
I am for dropping 1.7.10. We aim to support the latest server version, but also previous versions as a secondary objective.

If 1.7.10 issues are occurring as a result of the latest server version dropping API features, we should not be supporting that far back.

@Sirse Your best option, unfortunately, is to downgrade to a working version and be out of date. But I do not recommend this as a stable option because bug fixes may occur in the future, even critical ones.

**Comment by mdcfe** [comment]
1.7.10 is no longer supported. [Build 501](https://ci.drtshock.net/job/EssentialsX/501/) does not contain the fixes for 1.12, so should still work on 1.7.10.
